py4DSTEM/io/nonnative/read_dm.py

```diff
@@ -28,39 +29,70 @@ def read_dm(fp, mem="RAM", binfactor=1, metadata=False, **kwargs):
                     metadata is read and returned, in the former case a DataCube
                     is returned with the metadata attached at datacube.metadata
     """
-    assert(isinstance(fp,(str,Path))), "Error: filepath fp must be a string or pathlib.Path"
-    assert(mem in ['RAM','MEMMAP']), 'Error: argument mem must be either "RAM" or "MEMMAP"'
-    assert(isinstance(binfactor,int)), "Error: argument binfactor must be an integer"
-    assert(binfactor>=1), "Error: binfactor must be >= 1"
+    assert isinstance(
+        fp, (str, Path)
+    ), "Error: filepath fp must be a string or pathlib.Path"
+    assert mem in [
+        "RAM",
+        "MEMMAP",
+    ], 'Error: argument mem must be either "RAM" or "MEMMAP"'
+    assert isinstance(binfactor, int), "Error: argument binfactor must be an integer"
+    assert binfactor >= 1, "Error: binfactor must be >= 1"
 
     md = get_metadata_from_dmFile(fp)
     if metadata:
         return md
 
-    if (mem,binfactor)==("RAM",1):
-        with dm.fileDM(fp,on_memory=True) as dmFile:
-            dataSet = dmFile.getDataset(0)
-            dc = DataCube(data=dataSet['data'])
-    elif (mem,binfactor)==("MEMMAP",1):
+    if (mem, binfactor) == ("RAM", 1):
+        with dm.fileDM(fp, on_memory=True) as dmFile:
+            # loop through the datasets until a >2D one is found:
+            i = 0
+            valid_data = False
+            while not valid_data:
+                data = dmFile.getMemmap(i)
+                if len(np.squeeze(data).shape) > 2 :
+                    valid_data = True
+                    dataSet = dmFile.getDataset(i)
+                i += 1
+            dc = DataCube(data=dataSet["data"])
```

Comment:
This is a meaningful change.  Will this not break if we have a 3D dataset?  If we're going to add this step, why not check to see if the data is 4D?  Is this to support 4DSTEM .dm files which have unraveled (Rx,Ry) into one long dimension?  They changed that a while back, yes?  I guess I'm wondering if we need to keep supporting this..... :/  I suppose we probably do....what do you think?

Reply:
Aha yes this is a weird looking block, but the reason for it is that in the K3's consolidated `dm4` files, it saves a preview diffraction pattern with size `(1,1,Q_Nx,Q_Ny)`, hence the `squeeze`. 

In the Titan's `dm4` files, the data is saved as a 3D sequence, which (I have double-checked) is correctly chosen by this. These 3D datasets are still routinely produced.

Reply:
Gotcha.  Sigh!  Ok, looks good to me then